clang/include/clang/Sema/Sema.h

```diff
@@ -7810,9 +7813,11 @@ class Sema final : public SemaBase {
   }
 
   bool isInLifetimeExtendingContext() const {
-    assert(!ExprEvalContexts.empty() &&
```

Comment:
We check `isInLifetimeExtendingContext` when we create a `MaterializedTemporaryExpr`, if in a lifetime extend context, we will collect created MaterializedTemporaryExprs and extend its lifetime to a variable. 
e.g. (P2718R0 - Lifetime extension in range-based for loops.)In a for-range statement, we collect temporaries in for-range-init, and extend it's lifetime bound to for-range-var.

Reply:
Oh, I understand your concern. It's because we use `Sema::currentEvaluationContext()`, it's already has checked that `ExprEvalContexts` is not empty, so I think it's redundant,  and I removed assertion in `isInLifetimeExtendingContext`. Should we add the assertion back? 
```C++
  const ExpressionEvaluationContextRecord &currentEvaluationContext() const {
    assert(!ExprEvalContexts.empty() &&
           "Must be in an expression evaluation context");
    return ExprEvalContexts.back();
  };
```

Reply:
Ah!  I should have thought that was the case.  No need to add it back, thanks for clarifying!